extension/app/ts/simulation/protectors/commonTokenOops.ts

```diff
@@ -12,11 +12,12 @@ export const ADDITIONAL_BAD_TRANSFER_TARGETS = new Set<bigint>([
 ])
 
 export async function commonTokenOops(transaction: EthereumUnsignedTransaction, _simulator: Simulator) {
-	const transferInfo = getTransferInfoFromTx(transaction)
+	const transferInfo = parseTransaction(transaction)
 	if (transferInfo === undefined) return
 	if (transaction.to === null) return
-	if (!ADDITIONAL_BAD_TRANSFER_TARGETS.has(transferInfo.to)) return
-	if (tokenMetadata.get(addressString(transferInfo.to)) === undefined) return
-	if (nftMetadata.get(addressString(transferInfo.to)) === undefined) return
+	if (transferInfo.name !== 'transfer' && transferInfo.name !== 'transferFrom') return
+	if (!ADDITIONAL_BAD_TRANSFER_TARGETS.has(transferInfo.arguments.to)) return
+	if (tokenMetadata.get(addressString(transferInfo.arguments.to)) === undefined) return
+	if (nftMetadata.get(addressString(transferInfo.arguments.to)) === undefined) return
```

Comment:
Intentionally not using `ethers.getAddress(...)` here, or does it do something else?

Note: `addressString(...)` does *NOT* do checksumming.

Reply:
yeah, I am using non-checkummed, all lowercase addresses internally (or directly bigints) and checksummed ones for what is visible for user